packages/tools/viewer-alpha/src/viewer.ts

```diff
@@ -110,18 +160,121 @@ export class Viewer implements IDisposable {
         this._camera = new ArcRotateCamera("camera1", 0, 0, 1, Vector3.Zero(), this._details.scene);
         this._camera.attachControl();
         this._updateCamera(); // set default camera values
+        this._autoRotationBehavior = this._camera.getBehaviorByName("AutoRotation") as AutoRotationBehavior;
 
         // Load a default light, but ignore errors as the user might be immediately loading their own environment.
         this.loadEnvironmentAsync(undefined).catch(() => {});
 
         // TODO: render at least back ground. Maybe we can only run renderloop when a mesh is loaded. What to render until then?
-        this._engine.runRenderLoop(() => {
+        const render = () => {
             this._details.scene.render();
-        });
+            if (this.isAnimationPlaying) {
+                this.onAnimationProgressChanged.notifyObservers();
+                this._autoRotationBehavior.resetLastInteractionTime();
+            }
+        };
+
+        this._engine.runRenderLoop(render);
+        this._renderLoop = {

```

Comment:
Wondering about the naming ?

Reply:
You mean of the `_renderLoop` property? I'm just trying to store an `IDisposable` to make it easy to stop the render loop later. I think an an earlier iteration the render function was using some local state. In its current form it only uses class state, so I could just move it out to a private function of the class and directly call stopRenderLoop on it in the dispose. Or I could rename `_renderLoop` if you have suggestions on a better name.

Reply:
I have no preferences, and for the name, this is hard :-) renderLoop sounds like the render function I am not sure how to make sounds like it is the "stopper"

Reply:
Copilot suggested `_renderLoopController`. Going with it! :)

Reply:
Great one !!!